resources/timerbox.ts

```diff
@@ -1,136 +1,226 @@
-class TimerBox extends HTMLElement {
-  static get observedAttributes() {
+export default class TimerBox extends HTMLElement {
+  private _start: number;
+  private _duration: number;
+  rootElement: HTMLDivElement;
+  largeBoxElement: HTMLDivElement;
+  largeBoxBackgroundElement: HTMLDivElement;
+  largeBoxForegroundElement: HTMLDivElement;
+  smallBoxElement: HTMLDivElement;
+  smallBoxBackgroundElement: HTMLDivElement;
+  smallBoxForegroundElement: HTMLDivElement;
+  timerElement: HTMLDivElement;
+  readonly kLargeSize: number;
+  readonly kSmallSize: number;
+  readonly kBorderSize: number;
+  readonly kFontSize: number;
+  readonly kAnimateMS: number;
+  private _threshold: number;
+  private _bg: string;
+  private _fg: string;
+  private _scale: number;
+  private _valueScale: number;
+  private _towardTop: boolean;
+  private _styleFill: boolean;
+  private _hideAfter: number;
+  private _bigAtZero: boolean;
+  private _roundUpThreshold: boolean;
+  private _connected: boolean;
+  private _hideTimer: number | null;
+  private _timer: number | null;
+  private _animationFrame: number | null;
+
+  static get observedAttributes(): string[] {
```

Comment:
I think this needs styleFill as well, although it seems like setting this to empty is working correctly, whereas it is not for timer bars.